ignite/cmd/ignite/analytics.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/manifoldco/promptui"
+
+	"github.com/ignite/cli/ignite/pkg/gacli"
+	"github.com/ignite/cli/ignite/version"
+)
+
+const (
+	gaid          = "<GA_TRACK_ID>"
+	envDoNotTrack = "DO_NOT_TRACK"
+)
+
+var (
+	gaclient           *gacli.Client
+	igniteDir          = ".ignite"
+	igniteAnonIdentity = "anon"
+)
+
+type (
+	// metric represents an analytics metric.
+	metric struct {
+		// err sets metrics type as an error metric.
+		err error
+		// command is the command name.
+		command string
+	}
+
+	// identity represents an analytics identity file.
+	identity struct {
+		// name represents the user name.
+		Name string `json:"name" yaml:"name"`
+		// doNotTrack represents the user track choice.
+		DoNotTrack bool `json:"doNotTrack" yaml:"doNotTrack"`
+	}
+)
+
+func init() {
+	gaclient = gacli.New(gaid)
+}
+
+func addCmdMetric(m metric) {
+	envDoNotTrackVar := os.Getenv(envDoNotTrack)
+	if envDoNotTrackVar == "1" || strings.ToLower(envDoNotTrackVar) == "true" {
+		return
+	}
+
+	if m.command == "ignite version" {
+		return
+	}
+
+	ident, err := prepareMetrics()
+	if err != nil {
+		return
+	}
+
+	var met gacli.Metric
+	switch {
+	case m.err == nil:
+		met.Category = "success"
+	case m.err != nil:
+		met.Category = "error"
+		met.Value = m.err.Error()
+	}
+	met.Action = m.command
+
+	cmds := strings.Split(m.command, " ")
+	met.Label = cmds[0]
+	if len(cmds) > 0 {
+		met.Label = cmds[1]
+	}
+	met.User = ident.Name
+	met.Version = version.Version
```

Comment:
Maybe to group metric initialization values:

```suggestion
	met := gacli.Metric{
		Action: m.command,
		User: ident.Name,
		Version: version.Version,
	}

	switch {
	case m.err == nil:
		met.Category = "success"
	case m.err != nil:
		met.Category = "error"
		met.Value = m.err.Error()
	}

	cmds := strings.Split(m.command, " ")
	met.Label = cmds[0]
	if len(cmds) > 0 {
		met.Label = cmds[1]
	}
```